Northstar.CustomServers/mod/scripts/vscripts/mp/_stats.nut

```diff
@@ -888,30 +888,32 @@ void function OnWinnerDetermined()
 		player.SetPersistentVar( "kdratio_lifetime_pvp", kdratio_lifetimepvp )
 	}
 
-	// award mvp and top 3 in each team
-	if ( !IsFFAGame() )
+	array<entity> players = GetPlayerArray()
+	players.sort( GetScoreboardCompareFunc() )
+	int playerCount = players.len()
+	int currentPlace = 1
+	for ( int i = 0; i < 3; i++ )
 	{
-		string gamemode = GameRules_GetGameMode()
-		int functionref( entity, entity ) compareFunc = GameMode_GetScoreCompareFunc( gamemode )
-
-		for( int team = 0; team < MAX_TEAMS; team++ )
+		if ( i >= playerCount )
+			continue
+		
+		int functionref( entity, entity ) compareFunc = GetScoreboardCompareFunc()
+		if ( i > 0 && compareFunc( players[i - 1], players[i] ) != 0 )
+			currentPlace += 1
+		switch( currentPlace )
 		{
-			array<entity> players = GetPlayerArrayOfTeam( team )
-			if ( compareFunc == null )
-			{
-				printt( "gamemode doesn't have a compare func to get the top 3" )
-				return
-			}
-			players.sort( compareFunc )
-			int maxAwards = int( min( players.len(), 3 ) )
-			for ( int i = 0; i < maxAwards; i++ )
-			{
-				if ( i == 0 )
-					Stats_IncrementStat( players[ i ], "game_stats", "mvp", "", 1.0 )
-				Stats_IncrementStat( players[ i ], "game_stats", "top3OnTeam", "", 1.0 )
-			}
+			case 1:
+				UpdatePlayerStat( players[i], "game_stats", "mvp" ) // MVP in the current map played
```

Comment:
The comments here honestly confuse things more imo. (Other than the top3 one)

Reply:
Changes done